src/main/java/org/jabref/gui/BasePanel.java

```diff
@@ -767,12 +771,71 @@ private void cut() {
      *            "deleted". If true the action will be localized as "cut"
      */
     private void delete(boolean cut) {
-        List<BibEntry> entries = mainTable.getSelectedEntries();
+    	boolean bRemoveFromGroup = false;
+    	boolean bDelete = false;
+    	List<FieldChange> changesRemove = new ArrayList<>();
+    	
+    	List<BibEntry> entries = mainTable.getSelectedEntries();
+
         if (entries.isEmpty()) {
             return;
         }
-        if (!cut && !showDeleteConfirmationDialog(entries.size())) {
-            return;
+    	
+    	String selectedGroup = "";
+    	if (Globals.stateManager.getSelectedGroup(bibDatabaseContext).size() > 0) {
+    		selectedGroup = Globals.stateManager.getSelectedGroup(bibDatabaseContext).get(0).getName();
+    	}
+
+        //get list of all groups
+        Optional<GroupTreeNode> groups = bibDatabaseContext.getMetaData().getGroups();
```

Comment:
You have to check the optional either with ifPresent or isPresent to see if there is a value otherwise your next line will produce an NPE if there are no groups